jerry-ext/include/jerryscript-ext/module.h

```diff
@@ -34,32 +27,67 @@ typedef jerry_value_t (*jerryx_native_module_on_resolve_t) (void);
  * Declare the structure used to define a module. One should only make use of this structure via the
  * JERRYX_NATIVE_MODULE macro declared below.
  */
-typedef struct
+typedef struct jerryx_native_module_t
 {
   jerry_char_t *name; /**< name of the module */
   jerryx_native_module_on_resolve_t on_resolve; /**< function that returns a new instance of the module */
+  struct jerryx_native_module_t *next; /**< pointer to next module in the list */
 } jerryx_native_module_t;
 
 /**
- * Declare a helper macro that expands to the declaration of a variable of type jerryx_native_module_t placed into the
- * specially-named linker section "jerryx_modules" where the JerryScript module resolver
- * jerryx_module_native_resolver () will look for it.
+ * Declare the constructor and destructor attributes. These evaluate to nothing if this extension is built without
+ * library constructor/destructor support.
+ */
+#ifdef ENABLE_INIT_FINI
+#define JERRYX_MODULE_CONSTRUCTOR_ATTRIBUTE __attribute__((constructor))
+#define JERRYX_MODULE_DESTRUCTOR_ATTRIBUTE __attribute__((destructor))
+#define JERRYX_MODULE_REGISTRATION_QUALIFIER static
+#else /* !ENABLE_INIT_FINI */
+#define JERRYX_MODULE_CONSTRUCTOR_ATTRIBUTE
+#define JERRYX_MODULE_DESTRUCTOR_ATTRIBUTE
+#define JERRYX_MODULE_REGISTRATION_QUALIFIER
+#endif /* ENABLE_INIT_FINI */
+
+#define JERRYX_NATIVE_MODULE(module_name, on_resolve_cb)              \
+  static jerryx_native_module_t _##module_name##_definition =         \
+  {                                                                   \
+    .name = (jerry_char_t *) #module_name,                            \
```

Comment:
I think it was asked to not use hash-mark here in the old module system.

Reply:
We need it though, because we also have `_##module_name##_definition` and `module_name##_register()` and `module_name##_unregister()`.

Reply:
I found an interesting solution in node.